activejob/lib/active_job/exceptions.rb

```diff
@@ -44,14 +44,24 @@ module ClassMethods
       #  end
       def retry_on(*exceptions, wait: 3.seconds, attempts: 5, queue: nil, priority: nil)
         rescue_from(*exceptions) do |error|
+          payload = {
+            job: self,
+            adapter: self.class.queue_adapter,
+            error: error,
+            wait: wait
+          }
+
           if executions < attempts
-            logger.error "Retrying #{self.class} in #{wait} seconds, due to a #{error.class}. The original exception was #{error.cause.inspect}."
-            retry_job wait: determine_delay(wait), queue: queue, priority: priority
+            ActiveSupport::Notifications.instrument("enqueue_retry.active_job", payload) do
+              retry_job wait: determine_delay(wait), queue: queue, priority: priority
```

Comment:
Should `retry_job` publish this notification?

Reply:
That is a good question. My thought about this was that if you do the retry manually you don't need to be notified given you know you did the retry.

Reply:
Sure. I just think from a UX perspective it seems weird that calling `retry_job` doesn't publish the `enqueue_retry` notification.

Reply:
Should we move `ActiveSupport::Notifications.instrument("enqueue_retry.active_job", payload)` to `retry_job`?